wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -213,6 +218,19 @@ public <T, E extends Exception> T execute(
       final JdbcCallable<T, E> jdbcMethodFunc,
       final Object[] jdbcMethodArgs)
       throws E {
+
+    try {
+      if (this.enableFailoverSetting
+          && !METHOD_ABORT.equals(methodName)
+          && !METHOD_CLOSE.equals(methodName)
+          && this.pluginService.getCurrentConnection() != null
+          && this.pluginService.getCurrentConnection().isClosed()) {
+        this.pickNewConnection();
```

Comment:
I'm wondering if the condition for picking a new connection here should be stricter. For example, if the user is calling `isClosed` themselves, I don't think it makes sense to fail over